saos-import/src/test/java/pl/edu/icm/saos/importer/notapi/common/content/transaction/ContentFileCommitRollbackServiceTest.java

```diff
@@ -0,0 +1,136 @@
+    }
+    
+    @Test
+    public void commit_ADDED_FILES() throws IOException {
+        // given
+        ContentFileTransactionContext context = new ContentFileTransactionContext(contentDir, deletedTmpDir);
+        
+        context.addAddedFile("file1.txt");
+        context.addAddedFile("file2.txt");
+        
+        File file1 = new File(contentDir, "file1.txt");
+        File file2 = new File(contentDir, "file2.txt");
+        file1.createNewFile();
+        file2.createNewFile();
+        
+        // execute
+        contentFileCommitRollbackService.commit(context);
+        
+        // assert
+        assertTrue(file1.exists());
+        assertTrue(file2.exists());
+        assertFalse(deletedTmpDir.exists());
+    }
+    
+    @Test
+    public void commit_DELETED_FILES() throws IOException {
+        // given
+        ContentFileTransactionContext context = new ContentFileTransactionContext(contentDir, deletedTmpDir);
+        
+        File file1 = new File(deletedTmpDir, "file1.txt");
+        File file2 = new File(deletedTmpDir, "file2.txt");
+        file1.createNewFile();
+        file2.createNewFile();
+        
+        // execute
+        contentFileCommitRollbackService.commit(context);
+        
+        // assert
+        assertFalse(file1.exists());
+        assertFalse(file2.exists());
+        assertFalse(deletedTmpDir.exists());
+    }
+    
+    @Test(expected = NullPointerException.class)
+    public void rollback_NULL_CONTEXT() throws IOException {
+        contentFileCommitRollbackService.rollback(null);
+    }
+    
+    @Test
+    public void rollback_ADDED_FILES() throws IOException {
+        // given
+        ContentFileTransactionContext context = new ContentFileTransactionContext(contentDir, deletedTmpDir);
+        
+        context.addAddedFile("file1.txt");
+        context.addAddedFile("file2.txt");
+        
+        File file1 = new File(contentDir, "file1.txt");
+        File file2 = new File(contentDir, "file2.txt");
+        file1.createNewFile();
+        file2.createNewFile();
+        
+        // execute
+        contentFileCommitRollbackService.rollback(context);
+        
+        // assert
+        assertFalse(file1.exists());
+        assertFalse(file2.exists());
+        assertFalse(deletedTmpDir.exists());
+    }
+    
+    @Test
+    public void rollback_DELETED_FILES() throws IOException {
+        // given
+        ContentFileTransactionContext context = new ContentFileTransactionContext(contentDir, deletedTmpDir);
+        
+        File file1 = new File(deletedTmpDir, "file1.txt");
+        File file2 = new File(deletedTmpDir, "file2.txt");
+        file1.createNewFile();
+        file2.createNewFile();
+        
+        // execute
+        contentFileCommitRollbackService.rollback(context);
+        
+        // assert
+        assertTrue(new File(contentDir, "file1.txt").exists());
+        assertTrue(new File(contentDir, "file2.txt").exists());
```

Comment:
I would check here if the content of the corresponding files in delete directory and in content directory are equal.